Sources/_SwiftSyntaxTestSupport/SyntaxProtocol+Initializer.swift

```diff
@@ -23,7 +23,7 @@ private class InitializerExprFormat: BasicFormat {
   private func formatChildrenSeparatedByNewline<SyntaxType: SyntaxProtocol>(children: SyntaxChildren, elementType: SyntaxType.Type) -> [SyntaxType] {
     increaseIndentationLevel()
     var formattedChildren = children.map {
-      self.visit($0).as(SyntaxType.self)!
+      self.rewrite($0.cast(SyntaxType.self)).cast(SyntaxType.self)
```

Comment:
Why do we need the first `.cast(SyntaxType.self)`?

Reply:
This is because of the removal of `visit(Syntax)`. Personally I still think it's worth it so that we don't accidentally use `visit(Syntax)` instead of `rewrite(some SyntaxProtocol)` (the latter keeps the parent, the former doesn't necessarily). Thoughts?